### ISSUE TITLE
Add arm64 support for docker image

### ISSUE THREAD

**Issue by Ejmathewp** [hackthedev/dcts-shipping#29]
It would be great to enable multi-platform builds in your docker flow to allow running on arm64 (oracle free tier VPS for example).

**Comment by hackthedev** [comment]
Sounds interesting i will check it out!

**Comment by driftywinds** [comment]
Would like to bump this! This way I can host it on my pi 5

**Comment by t2vee** [comment]
@driftywinds Jeu @Ejmathewp  
I have implemented a few test builds of arm packages for dcts in my forked repository. 
Unfortunately I do not have access to a arm device to test these builds. Could one of you try it and see how it goes? It would be greatly appreciated 😊

you can find the image tags [here](https://github.com/t2vee/dcts-shipping/pkgs/container/dcts-shipping)

once validated i'll make a pull request into the main repo

**Comment by Ejmathewp** [comment]
Definitely! I might not be able to until Monday, but I'll spin it to and give it a try when I have some spare time.

**Comment by driftywinds** [comment]
> @driftywinds Jeu @Ejmathewp I have implemented a few test builds of arm packages for dcts in my forked repository. Unfortunately I do not have access to a arm device to test these builds. Could one of you try it and see how it goes? It would be greatly appreciated 😊
> 
> you can find the image tags [here](https://github.com/t2vee/dcts-shipping/pkgs/container/dcts-shipping)
> 
> once validated i'll make a pull request into the main repo

Will test and let you know!

**Comment by driftywinds** [comment]
> @driftywinds Jeu @Ejmathewp I have implemented a few test builds of arm packages for dcts in my forked repository. Unfortunately I do not have access to a arm device to test these builds. Could one of you try it and see how it goes? It would be greatly appreciated 😊
> 
> you can find the image tags [here](https://github.com/t2vee/dcts-shipping/pkgs/container/dcts-shipping)
> 
> once validated i'll make a pull request into the main repo

![image](https://github.com/user-attachments/assets/0112c202-a8e0-482a-b5ca-5adc398c37a5)

This image still gives me an exec format error

**Comment by Ejmathewp** [comment]
Hmm, I don't get a format error, but I get:
`exec /usr/local/bin/docker-entrypoint.sh: no such file or directory`

@driftywinds did you use the arm64 tag?
`docker pull ghcr.io/t2vee/dcts-shipping:latest@sha256:ad6486fc357ab61809ebe05a2887302ecd7360ac83415a10c012ba18e13a8358`

**Comment by driftywinds** [comment]
> Hmm, I don't get a format error, but I get: `exec /usr/local/bin/docker-entrypoint.sh: no such file or directory`
> 
> @driftywinds did you use the arm64 tag? `docker pull ghcr.io/t2vee/dcts-shipping:latest@sha256:ad6486fc357ab61809ebe05a2887302ecd7360ac83415a10c012ba18e13a8358`

Yes, I for sure used that very tag

**Comment by t2vee** [comment]
@driftywinds @Ejmathewp 
Thanks for testing those images! 
Clearly there is more work to do than I thought so ill be working on arm compatibility for the near future

**Comment by Cosmicoppai** [comment]
@t2vee can you share the Dockerfile and command used to build the image.

**Comment by driftywinds** [comment]
Any updates on the arm64 docker image?

**Comment by hackthedev** [comment]
@t2vee  if you have time @driftywinds  would need a bit of help

**Comment by Reeperk** [comment]
does it work now? @driftywinds

**Comment by driftywinds** [comment]
I will check and get back to you

**Comment by driftywinds** [comment]
<img width="552" height="30" alt="Image" src="https://github.com/user-attachments/assets/d0b523f8-8c78-44fc-a0ce-b00adfce4052" />

This is what I get when trying to pull arm64 image

**Comment by Reeperk** [comment]
Oh shit, I guess I forgot something. XD Sorry.
It should be fixed with https://github.com/hackthedev/dcts-shipping/pull/56, I hope.

Are you still using a Pi 5? If so, I can try it myself later and mention you when I’ve fixed it on my side.

**Comment by Reeperk** [comment]
so we should now have arm as a docker image in the workflow. https://github.com/hackthedev/dcts-shipping/pkgs/container/dcts-shipping

<img width="688" height="232" alt="Image" src="https://github.com/user-attachments/assets/ff4ee14b-814e-4848-9f95-0b3b69d91827" />

Can you test it @driftywinds?

**Comment by driftywinds** [comment]
@Reeperk works perfectly now!!

**Comment by hackthedev** [comment]
ur welcome was all my work hehe :D 




 (not)

**Comment by Reeperk** [comment]
perfect. Ig we can close it now @hackthedev  :)

**Comment by hackthedev** [comment]
> perfect. Ig we can close it now [@hackthedev](https://github.com/hackthedev) :)

hmmmmmmmmmmmmmmmmmmmm

**Comment by Cherryblue** [comment]
Hi, just tried to pull the last image (ghcr.io/hackthedev/dcts-shipping:latest) and i've got "The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested".

Looking at the repo, I tried instead "ghcr.io/hackthedev/dcts-shipping:9.0.2@sha256:c008f27ff0add10634e6a578f493d1d14dcc95e1558ade173313dd0e728ae77d" which works.

is the problem linked to "latest" release, or must-i always give the sha256 for it to identify correctly the arm build ?

**Comment by Reeperk** [comment]
> Hi, just tried to pull the last image (ghcr.io/hackthedev/dcts-shipping:latest) and i've got "The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested".

That's very funny, because we have that for latest. Docker probably didn't find it, but there is a parameter to force it. (--platform)

> is the problem linked to "latest" release, or must-i always give the sha256 for it to identify correctly the arm build ?

I'm not sure what you mean, but on GitHub you always have to specify the sha256 when pulling, as far as I know.


You can also (but I guess you already know that) see all the architectures here under the “OS / Arch” tab.
https://github.com/hackthedev/dcts-shipping/pkgs/container/dcts-shipping

and see all versions here:
https://github.com/hackthedev/dcts-shipping/pkgs/container/dcts-shipping/versions

**Comment by Cherryblue** [comment]
I have like 20 different self hosted services using multiple docker-compose ('been doing this for 5-10 years), and I had never filled sha in image field before, until today for dcts that is.

I do understand it's better for integrity reasons obviously, but it's far from a requirement. 
_EDIT : i do not know for github specifically, the major part of the images I pulled were from hub.docker.com, or linuxserver.io_

(And yeah that's what I did for the image alias with sha256. This did work, but since I saw nothing in the name specific for ARM appart from the sha256, I thought to ask here :))

**Comment by Reeperk** [comment]
> I have like 20 different self hosted services using multiple docker-compose ('been doing this for 5-10 years), and I had never filled sha in image field before, until today for dcts that is.

If you're using docker-compose, I agree with you: it's unusual to have to specify a sha256 there.

I've tested this myself again and have to correct myself here: GHCR doesn't require a sha256 digest (I had to use it once, but somehow it works out of nowhere). I can pull the image without any problems without sha256 – both with docker pull and via docker-compose.
So my earlier statement that you always have to specify a sha256 with GHCR was wrong – sorry about that.

> I do understand it's better for integrity reasons obviously, but it's far from a requirement. EDIT : i do not know for github specifically, the major part of the images I pulled were from hub.docker.com, or linuxserver.io

I agree with you. Pinning a digest is optional and mainly useful for reproducibility, but it is not a requirement—not even at GHCR. See my last answer.

> (And yeah that's what I did for the image alias with sha256. This did work, but since I saw nothing in the name specific for ARM appart from the sha256, I thought to ask here :))

What is still unclear to me is why Docker resolves this as linux/amd64 on your side.
On my end (I used my arm64 device for that, specifically a Raspberry Pi 4), the image is pulled correctly for ARM, without specifying --platform or a sha256 digest.

This therefore doesn’t look like a general ARM issue on our side, but rather like something specific to your setup. Why it doesn’t work for you is honestly a mystery to me, and unfortunately I can’t really help further with that.

**Comment by Cherryblue** [comment]
Welp my goal was to inform you in case it was still not resolved, but if it's really only on my side, who cares :D.
My end-goal was to use a specific version, but I didn't know where to find the given tags on github at first, so I used "latest".

For me, it's simpler to have the given version of an app used written in the compose file, so all in all, even if the behavior with latest is weird, it's not a real problem for me since it works when specifying a given version number. Maybe it's a docker cache problem linked to previous images downloaded ?

Anyway : no need to reopen the issue if it's only on my side since it's not a problem for me, and thanks for your time !
